src/gamestate/render.rs

```diff
@@ -46,18 +46,24 @@ impl Gamestate {
         {
             //3d rendering!!!
             let mut d2 = d.begin_mode3D(Camera::perspective(
+                // _ this needs to be coming from the player position
                 Vector3::new(
                     (time * 0.1).sin() * 100.0,
                     100.0,
                     (time * 0.1).cos() * 100.0,
                 ),
+                // self.map.spawns.choose(&mut rand::thread_rng()).unwrap().pos,
                 Vector3::zero(),
                 Vector3::new(0.0, 1.0, 0.0),
                 90.0,
             ));
             self.map.render(&mut d2, thread);
         }
 
+        // rendering top left game code and fps display
+        d.draw_text(&format!("Game Code: {}", self.code), 0, 0, 20, Color::WHITE);
+        d.draw_text(&format!("FPS: {}", unsafe { ffi::GetFPS() as u32 }), 0, 20, 20, Color::WHITE);
```

Comment:
There is a normal function that doesn’t need raylib ffi

Reply:
A normal function that gets the fps? I tried to use that but i ran into some errors so I just used the code the function is using